### ISSUE TITLE
Issue with deploying the app locally (1: Deploy the integration locally on your PC)

### ISSUE THREAD

**Issue by fstojanovic** [CSS-Electronics/canedge-grafana-backend#4]
**Commit : 196aa9efc4ca892554eabff52175e6b2523b3e6** (tag : v1.0.0)

**Platform : Windows 10**

I have followed these steps : https://github.com/CSS-Electronics/canedge-grafana-backend#1-deploy-the-integration-locally-on-your-pc

The application starts up fine and is validated by SimpleJson;
As soon as I attempt to import the dashboard dashboard-template-sample-data.json, the following error appears :

```
(env) C:\{SNIPPED/PATH/}canedge-grafana-backend>python canedge_datasource_cli.py "file:///%cd%/LOG" --port 8080 --limit 100
Mount path: file:///C:\{SNIPPED/PATH/}canedge-grafana-backend/LOG
Loaded DBs: canmod-gps
2022-04-27 08:42:36,438 - waitress - INFO - Serving on http://0.0.0.0:8080
2022-04-27 08:54:08,622 - canedge_datasource.annotations - WARNING - Failed to annotate: int() argument must be a string, a bytes-like object or a number, not 'dict'
2022-04-27 08:54:08,628 - canedge_datasource.annotations - WARNING - Failed to annotate: int() argument must be a string, a bytes-like object or a number, not 'dict'
2022-04-27 08:54:08,854 - canedge_datasource - ERROR - Exception on /query [POST]
Traceback (most recent call last):
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\canedge_datasource\query.py", line 130, in query_view
    return jsonify(query_cache(req_in))
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\flask_caching\__init__.py", line 952, in decorated_function
    rv = f(*args, **kwargs)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\canedge_datasource\query.py", line 117, in query_cache
    res = _query_time_series(req, start_date, stop_date)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\canedge_datasource\query.py", line 171, in _query_time_series
    return time_series_phy_data(fs=app.fs,
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\canedge_datasource\signal.py", line 202, in time_series_phy_data
    log_files = canedge_browser.get_log_files(fs, device, start_date=start_date, stop_date=stop_date,
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\canedge_browser\listing.py", line 152, in get_log_files
    selected_sessions = _bisect_list(
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\canedge_browser\listing.py", line 354, in _bisect_list
    start_index = bisect.bisect_left(bisect_list, lower_bound)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\canedge_browser\support\FuncBackedList.py", line 39, in __getitem__
    self._values[item] = self._func(key)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\canedge_browser\support\FuncBackedList.py", line 15, in <lambda>
    self._func = lambda x: func(x, *args, **kwargs)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\canedge_browser\listing.py", line 325, in _extract_date_from_session_wrapper
    result = extract_date(handle, passwords)
  File "C:\{SNIPPED/PATH/}canedge-grafana-backend\env\lib\site-packages\canedge_browser\listing.py", line 239, in _extract_date_mdf4
    mdf_file = mdf_iter.MdfFile(*args)
TypeError: int() argument must be a string, a bytes-like object or a number, not 'dict'
```

**Comment by MatinF** [comment]
Hi, thanks for reporting.

Can you try and record the steps you take in particular within Grafana incl. your settings page for the SimpleJson integration?

I checked the same commit just now on Win 10 without issues.

best,
Martin

**Comment by MatinF** [comment]
We also made some updates to the `requirements.txt` - can you try doing a clean install with that to see if it resolves your issue?

**Comment by fstojanovic** [comment]
Hi MatinF, thanks for the update. The updated requirements.txt helped me to find the issue.

For whatever reason, my pip refused to install mdf-iter v0.0.7, and would complain when I attempted to upgrade it (no upgrade available). However, clearly the latest version is the v0.0.7.
The fix was to uninstall mdf-iter v0.0.4 completely, and then reinstall the latest version.
